core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -139,7 +144,8 @@ protected CqlRequestHandler(
       String sessionLogPrefix) {
 
     this.startTimeNanos = System.nanoTime();
-    this.logPrefix = sessionLogPrefix + "|" + this.hashCode();
+    this.distributedTraceIdGenerator = context.getDistributedTraceIdGenerator();
+    this.logPrefix = this.distributedTraceIdGenerator.getSessionRequestId(statement);
```

Comment:
Can we maintain the `logPrefix` logic in default `NoopDistributedTraceIdGenerator`? This prefix is inserted into logs, so it may be nice to preserve backward compatibility.